tbx/core/blocks.py

```diff
@@ -440,12 +440,24 @@ class BlogChooserBlock(blocks.StructBlock):
         max_num=3,
     )
 
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        context["is_standard_page"] = False
+        return context
+
     class Meta:
         icon = "link"
         template = "patterns/molecules/streamfield/blocks/blog_chooser_block.html"
         group = "Custom"
 
 
+class BlogChooserStandardPageBlock(BlogChooserBlock):
```

Comment:
@nickmoreton let me know if there's a better approach than subclassing the block here - I basically needed a way for the block to know in the template context whether it was on the standard page or not.

Reply:
Hi @helenb There probably is a more pythonic way to do this but it would need more code overall I think.

You could override the __init__ method of the base struct blocks and provide the true or false value there but you still need to get it into the the context for the template. So I think your method works well and is clear to others what the intention is.

Reply:
thanks :)